data/augmentation.py

```diff
@@ -0,0 +1,99 @@
+import random
+
+import numpy as np
+from tqdm import tqdm
+from datasets import Dataset
+from omegaconf import DictConfig
+
+from data.quantizer import MidiQuantizer
+
+
+def change_speed(record: dict, factor: float = None) -> dict:
+    if not factor:
+        slow = 0.8
+        change_range = 0.4
+        factor = slow + random.random() * change_range
+
+    record["start"] = [value / factor for value in record["start"]]
+    record["end"] = [value / factor for value in record["end"]]
+    record["duration"] = [x - y for x, y in zip(record["end"], record["start"])]
+    return record
+
+
+def pitch_shift(pitch: list[int], shift_threshold: int = 5) -> list[int]:
+    # No more than given number of steps
+    PITCH_LOW = 21
+    PITCH_HI = 108
+    low_shift = -min(shift_threshold, min(pitch) - PITCH_LOW)
+    high_shift = min(shift_threshold, PITCH_HI - max(pitch))
+
+    if low_shift > high_shift:
+        shift = 0
+    else:
+        shift = random.randint(low_shift, high_shift + 1)
+    pitch = [value + shift for value in pitch]
+
+    return pitch
+
+
+def apply_augmentation(record: dict, quantizer: MidiQuantizer, augmentation_probability: float):
+    augmented = record.copy()
+
+    # check if augmentation happened
+    done = False
+    # shift pitch augmentation
+    if random.random() < augmentation_probability:
+        # max shift is octave down or up
+        shift = random.randint(1, 12)
+        augmented["pitch"] = pitch_shift(augmented["pitch"], shift)
```

Comment:
Looks like you maybe misunderstood what `pitch_shift` is doing, and what the role `shift_threshold` is (it would be easier to spot if you used named parameters ;))

`shift_threshold` is controlling the maximum random pitch shift, there's no point in setting it randomly, as you could just set it to 12 here and get the same behaviour. 

The idea is that pitch shifting music too much is doing to much damage to the musical content (i.e. chopin shifted 2 octaves would be ridiculous), and we should treat it as a hyper parameter that we'll try to figure out based on the results we see.